ros2msg/test/test_cli.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+
+
+def test_cli():
+    packages_cmd = ['ros2', 'msg', 'packages']
+    packages_result = subprocess.run(packages_cmd, stdout=subprocess.PIPE, check=True)
+    package_names = packages_result.stdout.decode().splitlines()
+
+    # explicit dependencies of this package will for sure be available
+    assert 'std_msgs' in package_names
+
+    count = 0
+    for package_name in package_names:
+        package_cmd = ['ros2', 'msg', 'package', package_name]
```

Comment:
The advantage I saw with having separate "verbs" is that each of them is self contained. Additional ones can easily be added (without fiddling with existing code, e.g. by external Python packages). Also with a single "verb" handling multiple different cases you have to "read" the usage to figure out which option can be used together with others and how it might affect the output (e.g. the `--packages` and `--package` options would be mutually exclusive).